From 92ced30b60b38a20747360c519dde7d12618f368 Mon Sep 17 00:00:00 2001
From: "Wu, Hao" <hao.wu@intel.com>
Date: Thu, 10 Oct 2013 12:39:36 +0800
Subject: [PATCH 078/429] dwc_otg: move gpio request back to Intel dwc byt
 file.

Request GPIO in pci early quirk fixup function always failed
as it is earlier than GPIO init. so move it back to dwc otg
platform init function invoked during pci probe function.

Signed-off-by: Wu, Hao <hao.wu@intel.com>
---
 .../intel-mid/device_libs/pci/platform_usb_otg.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/arch/x86/platform/intel-mid/device_libs/pci/platform_usb_otg.c b/arch/x86/platform/intel-mid/device_libs/pci/platform_usb_otg.c
index 1a248252ac2a..5a943edf7ae7 100644
--- a/arch/x86/platform/intel-mid/device_libs/pci/platform_usb_otg.c
+++ b/arch/x86/platform/intel-mid/device_libs/pci/platform_usb_otg.c
@@ -24,6 +24,22 @@ static struct intel_dwc_otg_pdata *get_otg_platform_data(struct pci_dev *pdev)
 	case PCI_DEVICE_ID_INTEL_MRFL_DWC3_OTG:
 		if (intel_mid_identify_sim() == INTEL_MID_CPU_SIMULATION_HVP)
 			dwc_otg_pdata.is_hvp = 1;
+			dwc_otg_pdata.charger_detect_enable = 0;
+		}
+		return &dwc_otg_pdata;
+	case PCI_DEVICE_ID_INTEL_BYT_OTG:
+		/* FIXME: Hardcode now, but need to use ACPI table for GPIO */
+		if (INTEL_MID_BOARD(3, TABLET, BYT, BLK, PRO, RVP3) ||
+			INTEL_MID_BOARD(3, TABLET, BYT, BLK, ENG, RVP3)) {
+			pr_info("This is BYT RVP\n");
+			dwc_otg_pdata.gpio_cs = 156;
+			dwc_otg_pdata.gpio_reset = 144;
+		} else if (INTEL_MID_BOARD(3, TABLET, BYT, BLK, PRO, 8PR0) ||
+			INTEL_MID_BOARD(3, TABLET, BYT, BLK, ENG, 8PR0)) {
+			pr_info("This is BYT FFRD8 PR0\n");
+			dwc_otg_pdata.gpio_cs = 54;
+			dwc_otg_pdata.gpio_reset = 144;
+		}
 		return &dwc_otg_pdata;
 	default:
 		break;
-- 
2.37.3

